content/en/security/code_security/software_composition_analysis/setup_static/_index.md

```diff
@@ -285,71 +285,141 @@ If you already have automatic scanning enabled for a repository, a manual upload
-files with the runtime service. For example, if a service called `foo` has
-a log entry or a stack trace containing a file with a path `/modules/foo/bar.py`,
-it associates files `/modules/foo/bar.py` to service `foo`.
-{{% /collapse-content %}}
+The schema for this field is described in the [Software Catalog entity model][23].
 
-{{% collapse-content title="Detecting service name in paths and repository names" level="h4" %}}
+Datadog will go through all your Software Catalog definitions and check whether your finding’s file path matches. For a finding to be mapped to a service through `codeLocations`, it must contain a file path. Some findings may not contain a file path; in those cases, Datadog cannot evaluate `codeLocations` for that finding, and this mechanism is skipped.
 
-Datadog detects service names in paths and repository names, and associates the file with the service if a match is found.
+<div class="alert alert-danger">
+Services defined with a **Software Catalog schema v2.x** do **not** support `codeLocations`. Existing definitions can be upgraded to the v3 schema in the [Software Catalog][24]. Once the migration is complete, changes may take up to 24 hours to apply to findings. If you are unable to upgrade to v3, Datadog will fall back to alternative linking techniques (described below). These rely on less precise heuristics, so accuracy may vary depending on the Code Security product and your use of other Datadog features.
+</div>
 
-For a repository match, if there is a service called `myservice` and
-the repository URL is `https://github.com/myorganization/myservice.git`, then,
-it associates `myservice` to all files in the repository.
+#### Example (v3 schema)
 
-If no repository match is found, Datadog attempts to find a match in the
-`path` of the file. If there is a service named `myservice`, and the path is `/path/to/myservice/foo.py`, the file is associated with `myservice` because the service name is part of the path. If two services are present
-in the path, the service name closest to the filename is selected.
-{{% /collapse-content %}}
+{{< code-block lang="yaml" filename="entity.datadog.yaml" collapsible="true" >}}
+apiVersion: v3
+kind: service
+metadata:
+  name: billing-service
+  owner: billing-team
+datadog:
+  codeLocations:
+    - repositoryURL: https://github.com/org/myrepo.git
+      paths:
+        - path/to/service/code/**
+    - repositoryURL: https://github.com/org/billing-service.git
+      paths:
+        - "**"
+{{< /code-block >}}
+
+#### SAST finding
+
+If a vulnerability appeared in `github.com/org/myrepo` at `/src/billing/models/payment.py`, then using the `codeLocations` for `billing-service` Datadog would add `billing-service` as an owning service. If your service defines an `owner` (see above), then Datadog will link that team to the finding too. In this case, the finding would be linked to the `billing-team`.
```

Comment:
```suggestion
If a vulnerability appeared in `github.com/org/myrepo` at `/src/billing/models/payment.py`, then using the `codeLocations` for `billing-service` makes Datadog add `billing-service` as an owning service. If your service defines an `owner` (see above), then Datadog links that team to the finding too. In this case, the finding would be linked to the `billing-team`.
```

Reply:
@gorkavicente I'm not 100% sure I understand this one but I edited it for clarity. Can you confirm that it is accurate? thx